### PR TITLE
Fix model name not properly parsed by inference graph

### LINKED ISSUE #3526: In the inference graph, does the serviceName field require the inference service name to be the same as the model name?
I have two inference services: one is called diffusers-test with a model name of sdxl-turbo, and the other is called emotions-test with a model name of emotions.

I deploy a simple inferece graph as follows:
```
apiVersion: "serving.kserve.io/v1alpha1"
kind: "InferenceGraph"
metadata:
  name: "simple-pipeline"
spec:
  nodes:
    root:
      routerType: Sequence
      steps:
      - serviceName: emotions-test
        name: emotions-test # step name
      - serviceName: diffusers-test
        name: diffusers-test
        data: $response
```

Then I try to test the inference graph by following python code, which is simply sending a request
```python
import requests
import time
import json


url = "http://aaa.elb.amazonaws.com"
headers = {
    "Content-Type": "application/json",
    "Host": "simple-pipeline.default.example.com"
}
with open('./input.json', 'r') as file:
    data = json.load(file)
    data = json.dumps(data)

a=time.time()
try:
    start = time.time()
    print("sending request...")
    response = requests.post(url, headers=headers, data=data, timeout=24*60*60)
    end = time.time()
    print("请求耗时：", end - start, "秒")
    print("服务器响应：", response.text)
except requests.exceptions.Timeout:
    print("请求超时。服务器没有在预定时间内响应。")
except requests.exceptions.RequestException as e:
    print("请求失败：", e)
b=time.time()
print("总耗时：", b - a, "秒")
```

### PR DESCRIPTION
**Pull request by sivanantha321**
**What this PR does / why we need it**:
When model name is passed as an argument to the model server with the following format, 
- Args: ["--model_name", "iris"]
- ~Args: ["--model_name iris"]~

The inference graph controller fails to parse them and uses the inferenceservice name as the model name which leads to `404 model not found` error.
As of now, The only supported format which is recognized by the inference graph controller is
- Args: ["--model_name=iris"]

This PR fixes this issue and adds support for parsing the model name from the following formats.
- Args: ["--model_name=iris"]
- Args: ["--model_name", "iris"]
> - ~Args: ["--model_name iris"]~
   This format is not parsed by the model server, So ignoring this format.
- Args: ["--model_name", "isvc-iris", --model_name="iris"] - If multiple `--model_name` arguments present, The last argument value will be used.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
- Fixes #3526

**Type of changes**
Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Feature/Issue validation/testing**:

Please describe the tests that you ran to verify your changes and relevant result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

- Logs

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Checklist**:

- [x] Have you added unit/e2e tests that prove your fix is effective or that this feature works?
- [x] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

**Release note**:

```release-note
Fix model name not properly parsed by inference graph
```

**Re-running failed tests**

- `/rerun-all` - rerun all failed workflows.
- `/rerun-workflow <workflow name>` - rerun a specific failed workflow. Only one workflow name can be specified. Multiple /rerun-workflow commands are allowed per comment.

### CODE REVIEW

File: pkg/controller/v1beta1/inferenceservice/utils/utils.go [hunk 1]
```diff
@@ -85,18 +84,35 @@ func IsMemoryResourceAvailable(isvc *v1beta1api.InferenceService, totalReqMemory
...
 
+func getModelNameFromArgs(args []string) string {
+	for i, arg := range args {
+		if strings.HasPrefix(arg, constants.ArgumentModelName) {
+			// Case 1: ["--model-name=<model-name>"]
+			modelNameValueArr := strings.Split(arg, "=")
+			if len(modelNameValueArr) == 2 {
+				return modelNameValueArr[1]
+			}
+			// Case 2: ["--model-name <model-name>"]
```
Comment (israel-hdez) [inline pkg/controller/v1beta1/inferenceservice/utils/utils.go]: Do you want to tolerate more than one space in between? If so, you may need something more complex than a Split.

Comment (sivanantha321) [reply pkg/controller/v1beta1/inferenceservice/utils/utils.go]: Addressed.

**Review by spolti — APPROVED** [review]
/lgtm

File: pkg/apis/serving/v1beta1/predictor_tfserving.go [hunk 2]
```diff
@@ -28,13 +28,12 @@ import (
 )
 
 var (
-	TensorflowEntrypointCommand          = "/usr/bin/tensorflow_model_server"
-	TensorflowServingGRPCPort            = "9000"
-	TensorflowServingRestPort            = "8080"
-	TensorflowServingGPUSuffix           = "-gpu"
-	InvalidTensorflowRuntimeVersionError = "Tensorflow RuntimeVersion must be one of %s"
```
Comment (yuzisun) [inline pkg/apis/serving/v1beta1/predictor_tfserving.go:35]: `InvalidTensorflowRuntimeVersionError` is not used anymore right?

File: pkg/controller/v1beta1/inferenceservice/utils/utils.go [hunk 3]
```diff
@@ -85,18 +84,31 @@ func IsMemoryResourceAvailable(isvc *v1beta1api.InferenceService, totalReqMemory
...
 }
 
+func getModelNameFromArgs(args []string) string {
+	modelName := ""
+	for i, arg := range args {
+		if strings.HasPrefix(arg, constants.ArgumentModelName) {
+			// Case 1: ["--model-name=<model-name>"]
+			modelNameValueArr := strings.Split(arg, "=")
+			if len(modelNameValueArr) == 2 {
+				modelName = modelNameValueArr[1]
```
Comment (yuzisun) [inline pkg/controller/v1beta1/inferenceservice/utils/utils.go:94]:
```suggestion
				modelName = modelNameValueArr[1]
				return modelName
```

Comment (yuzisun) [reply pkg/controller/v1beta1/inferenceservice/utils/utils.go:94]: When both formats exist, do we expect to get the model name from the first or last arg?

Comment (yuzisun) [reply pkg/controller/v1beta1/inferenceservice/utils/utils.go:94]: from the test looks like you are expecting to get the model name from the last arg

Comment (sivanantha321) [reply pkg/controller/v1beta1/inferenceservice/utils/utils.go:94]: yeah, that's how python's argparse behaves.

File: pkg/apis/serving/v1beta1/predictor_tfserving.go (hunk 2 above)
Comment (sivanantha321) [reply pkg/apis/serving/v1beta1/predictor_tfserving.go:35]: yes

**Pull request merged**